docs/core/howto/ssl.rst

```diff
@@ -204,10 +199,40 @@ For example,
     options = CertificateOptions(..., dhParameters=dhParams)
 
 Another part of the TLS protocol which ``CertificateOptions`` can control is the version of the TLS or SSL protocol used.
-This is often called the context's "method".
+There are two ways to configure it, either by configuring a list of protocols, or by the legacy way of setting the internal context's "method".
+By default, Twisted will configure it to use TLSv1.0 or later and disable the insecure SSLv3 protocol, but this manual control over protocols can be helpful if you need to support legacy SSLv3 systems, or you wish to restrict it down to just the strongest of the TLS versions.
+
+You can give ``CertificateOptions`` a range of TLS versions you wish it to negotiate with the ``tlsProtocols`` argument in the initializer:
+
+.. code-block:: python
+
+    from twisted.internet.ssl import (CertificateOptions, TLSVersion,
+                                      TLSVersionRange)
+    options = CertificateOptions(
+        ...,
+        tlsProtocols=TLSVersionRange(TLSVersion.TLSv1_0,
+                                     TLSVersion.TLSv1_2))
+
```

Comment:
Originally, I used a set based API, but @Lukasa discouraged that -- one of the reasons that swayed me was that it binds us to OpenSSL (Windows' SChannel and possibly others accept a range in their API, vs particular protocols). I am honestly not sure what's better.

An alternate API would be possibly a `minimumTLSVersion` and a `maximumTLSVersion` with defaults of TLSv1.0 and a 'highest' canary respectively, and replace the concrete type and maybe make the API better? Then you can set just `minimumTLSVersion` if you want like, 1.2+, and not accidentally lock yourself out of future versions...